vllm/lora/punica.py

```diff
@@ -1,206 +1,260 @@
 # Based on code from https://github.com/punica-ai/punica
 
-from typing import Optional
+from typing import Dict, Optional, Tuple
 
 import torch
 
-from vllm import _custom_ops as ops
+from vllm.lora.ops.bgmv_expand import bgmv_expand
+from vllm.lora.ops.bgmv_expand_slice import bgmv_expand_slice
+from vllm.lora.ops.bgmv_shrink import bgmv_shrink
+from vllm.lora.ops.sgmv_expand import sgmv_expand
+from vllm.lora.ops.sgmv_expand_slice import sgmv_expand_slice
+from vllm.lora.ops.sgmv_shrink import sgmv_shrink
 
+_PARAMS_CACHE: Dict[int, Tuple] = {}
```

Comment:
ideally the this cache would be an attribute of LoRAModelManager, instead of a global (functions can take it in as an argument).

Reply:
I am worried that this will be hard to clean up/leak memory if it's left as a global

Reply:
Perhaps we should make this cache a static variable  would that be better?

 There's also an alternative solution: based on the discussion below, we'll need to refactor the return values of the `convert_mapping`  in the future,  and we should be able to do this together, implement a class similar to `FlashAttentionMetadata`.

I appreciate your guidance and feedback. Thank you very much.

Reply:
@Yard1 I have refactored this part of the code, please review it again， thanks~